src/libstore/tests/common-protocol.cc

```diff
@@ -20,61 +20,43 @@ class CommonProtoTest : public ProtoTest<CommonProto, commonProtoDir>
      * Golden test for `T` reading
      */
     template<typename T>
-    void readTest(PathView testStem, T value)
+    void readProtoTest(PathView testStem, T expected)
     {
-        if (testAccept())
-        {
-            GTEST_SKIP() << cannotReadGoldenMaster;
-        }
-        else
-        {
-            auto encoded = readFile(goldenMaster(testStem));
-
+        CharacterizationTest::readTest(testStem, [&](auto encoded) {
             T got = ({
                 StringSource from { encoded };
                 CommonProto::Serialise<T>::read(
                     *store,
                     CommonProto::ReadConn { .from = from });
             });
 
-            ASSERT_EQ(got, value);
-        }
+            ASSERT_EQ(got, expected);
+        });
     }
 
     /**
      * Golden test for `T` write
      */
     template<typename T>
-    void writeTest(PathView testStem, const T & value)
+    void writeProtoTest(PathView testStem, const T & decoded)
     {
-        auto file = goldenMaster(testStem);
-
-        StringSink to;
-        CommonProto::write(
-            *store,
-            CommonProto::WriteConn { .to = to },
-            value);
-
-        if (testAccept())
-        {
-            createDirs(dirOf(file));
-            writeFile(file, to.s);
-            GTEST_SKIP() << updatingGoldenMaster;
-        }
-        else
-        {
-            auto expected = readFile(file);
-            ASSERT_EQ(to.s, expected);
-        }
+        CharacterizationTest::writeTest(testStem, [&]() -> std::string {
+            StringSink to;
+            CommonProto::Serialise<T>::write(
+                *store,
+                CommonProto::WriteConn { .to = to },
+                decoded);
+            return std::move(to.s);
```

Comment:
```suggestion
            return to.s;
```

Please basically never write `return std::move(...);`. The compiler will do it automatically anyway, but certain optimizations can't take part any longer if this is done manually.

Reply:
Oops yes